enhancements/service-catalog/service-catalog-and-broker-removal.md

```diff
@@ -0,0 +1,187 @@
+---
+title: service-catalog-and-broker-removal
+authors:
+  - "@jmrodri"
+reviewers:
+  - "@dmesser"
+  - "@joelanford"
+  - "@robszumski"
+approvers:
+  - TBD
+creation-date: 2019-12-17
+last-updated: 2019-12-17
+status: implementable
+see-also:
+  - "/enhancements/service-catalog/prepare-service-catalog-and-brokers-for-removal.md"
+replaces:
+superseded-by:
+---
+
+# Service Catalog and Broker Removal
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+This is where to call out areas of the design that require closure before deciding
+to implement the design.  For instance, 
+ > 1. This requires exposing previously private resources which contain sensitive
+  information.  Can we do this? 
+
+## Summary
+
+In OpenShift 4.1, we announced the deprecation of the Service Catalog, the
+Ansible Service Broker, and the Template Service Broker. In 4.4, we plan to
+remove them from productization. They will no longer be installed in a new
+cluster. They will also block cluster upgrades until they are disabled in older
+clusters.
+
+## Motivation
+
+Service Catalog and Service Brokers need to be removed in OpenShift 4.4 because
+it is an LTS release and it is our desire avoid the extended LTS support period
+for these deprecated operators and services.
+
+### Goals
+
+1. Implement a pre-flight check for the CVO that ensures the Service Catalog and
+   Service Brokers are not present after an upgrade to OpenShift 4.4. (ENG)
+1. Alerts users during upgrade if Service Catalog and Service Brokers are in
+   use. (ENG)
+1. Prevent users from upgrading to OpenShift 4.4 if the Service Catalog or the
+   Service Brokers are in use. (ENG)
+1. Remove the Service Catalog from the OpenShift 4.4 release payload (ART)
+1. Remove the Ansible Service Broker from the OpenShift 4.4 release payload (ART)
+1. Remove the Template Service Broker from the OpenShift 4.4 release payload (ART)
+1. Remove the Template Service Broker from productization (ART)
+1. Remove the Ansible Service Broker from productization (ART)
+1. Remove the Template Service Broker Operator from productization (ART)
+1. Remove the Ansible Service Broker Operator from productization (ART)
+1. Remove the Service Catalog from productization (ART)
+
+### Non-Goals
+
+N/A
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+Create a pre-flight check in the CVO process to ensure the Service Catalog,
+Ansible Service Broker, Template Service Broker, or any broker registered to
+Service Catalog.
+
+During the upgrade, alert users if the Service Catalog or any Service Brokers
+are in use.
+
+#### Story 2
+
+Remove Service Catalog and the brokers from the release payload
```

Comment:
Seems like one simple option to handling this is to leave the service catalog operator present in 4.4, but update it to always remove its resources.  This would ensure everything is cleaned up and the CVO is only responsible for cleaning resources the CVO created.

Reply:
Updated proposal to include keeping the operators present.